src/provisioner/provisioner.py

```diff
@@ -101,7 +100,14 @@ def config_apply(solution_config_url: str, cortx_conf_url: str = None,
         if Conf.get(CortxProvisioner._solution_index, 'cluster') is not None:
             CortxProvisioner.apply_cluster_config(cortx_conf, CortxProvisioner.cortx_release)
 
-        if Conf.get(CortxProvisioner._solution_index, 'cortx') is not None:
+        if not CortxProvisioner._wait_for_lock_to_be_released(cortx_conf, CortxProvisioner._lock_timeout):
+            Conf.unlock(cortx_conf._conf_idx, lock_owner=Conf.machine_id, force = True, lock_key=CortxProvisioner._lock_key)
+            # TODO: remove Conf.save once gconf is completly moved to consul
+            Conf.save(cortx_conf._conf_idx)
+        if cortx_conf.get('cortx') is None and Conf.get(CortxProvisioner._solution_index, 'cortx') is not None:
```

Comment:
Please confirm if this condition is correct?

Reply:
correct as in? Yes it is correct as we are applying cortx config there.